### PR TITLE
Fix for GEOS-8128 - SLD static legend dimensions not respected

### PR DESCRIPTION
**Pull request by afabiani**

### CODE REVIEW

File: src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java [hunk 1]
```diff
@@ -248,6 +246,12 @@ private LegendRequest addLayer(LayerInfo layerInfo, GetLegendGraphicRequest requ
             LegendInfo legendInfo = resolveLegendInfo(layerInfo.getLegend(),request);
             if(legendInfo != null ){
```
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java]: This block of code is now repeated 3 times in different places, could it be refactored into a method call?

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java [hunk 2]
```diff
@@ -108,13 +108,23 @@ public void testCustomLegend() throws Exception {
         BufferedImage expected = ImageIO.read( resource.file() );
         
         assertEquals( getPixelColor(expected,10,2).getRGB(), getPixelColor(image,10,2).getRGB() );
+
+        // test external image dimensions
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java]: So the test checks the image size and not the value set in the setup... I suppose they are the same? Could be useful to have a test where they are not.

Comment (afabiani) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java]: @aaime Done!

File: src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java (hunk 1 above)
Comment (afabiani) [inline src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java]: @aaime Done!

**Pull request merged**